syne_tune/optimizer/schedulers/searchers/bore/bore.py

```diff
@@ -122,6 +127,11 @@ def __init__(
 
         self._hp_ranges = make_hyperparameter_ranges(self.config_space)
         self._excl_list = ExclusionList.empty_list(self._hp_ranges)
+        self._allow_duplicates = allow_duplicates
+        # Maps ``trial_id`` to configuration. This is used to blacklist
```

Comment:
This is a technicality:
- If not `allow_duplicates`, then `_excl_list` collects all configs suggested, so none is suggested 2x
- If `allow_duplicates`, then `_excl_list` only collects configs for which the trial failed. In this case, we need to know the config for a trial, because `evaluation_failed` only gets the `trial_id`.


The same code is used in other places, same comment applies.